lib/frontend/src/ssa.rs

```diff
@@ -39,7 +38,7 @@ where
 {
     // Records for every variable and for every revelant block, the last definition of
     // the variable in the block.
-    variables: EntityMap<Variable, HashMap<Block, Value>>,
+    variables: EntityMap<Variable, EntityMap<Block, Value>>,
```

Comment:
Should this be `EntityMap<Block, PackedOption<Value>>`? That gives you a sensible default value.